hibernation-setup-tool.c

```diff
@@ -1659,52 +1659,6 @@ static void link_hook(const char *src, const char *dest)
     spawn_and_wait("systemctl", 1, "daemon-reload");
 }
 
-static void ensure_systemd_hooks_are_set_up(void)
-{
-    /* Although the systemd manual cautions against dropping executables or scripts in
-     * this directory, the proposed D-Bus interface (Inhibitor) is not sufficient for
-     * our use case here: we're not trying to inhibit hibernation/suspend, we're just
-     * trying to know when this happened.
-     *
-     * More info: https://www.freedesktop.org/software/systemd/man/systemd-suspend.service.html
-     */
-    const char *execfn = (const char *)getauxval(AT_EXECFN);
-    const char *location_to_link = "/usr/lib/systemd/systemd-sleep";
-    struct stat st;
-    int r;
-
-    r = stat(location_to_link, &st);
-    if (r < 0 && errno == ENOENT) {
-        log_info("Attempting to create hibernate/resume hook directory: %s", location_to_link);
-        if (mkdir(location_to_link, 0755) < 0) {
-            log_info("Couldn't create %s: %s. VM host won't receive suspend/resume notifications.", location_to_link, strerror(errno));
-            return;
-        }
-    } else if (r < 0) {
-        log_info("Couldn't stat(%s): %s. We need to drop a file there to allow "
-                 "the VM host to be notified of hibernation/resumes.",
-                 location_to_link, strerror(errno));
-        return;
-    } else if (!S_ISDIR(st.st_mode)) {
-        log_info("%s isn't a directory, can't drop a link to the agent there to "
-                 " notify host of hibernation/resume",
-                 location_to_link);
-        return;
-    }
-
-    if (execfn)
-        return link_hook(execfn, location_to_link);
-
-    char self_path_buf[PATH_MAX];
-    const char *self_path = readlink0("/proc/self/exe", self_path_buf);
-    if (self_path)
-        return link_hook(self_path, location_to_link);
```

Comment:
Can you please add a description in the commit message to explain why ensure_systemd_hooks_are_set_up() should be removed? link_hook() is only used in ensure_systemd_hooks_are_set_up(). Should we remove link_hook() as well?

Reply:
Currently git commit message for this removal is "removed deprecated approach to sleep hooks" (https://github.com/microsoft/hibernation-setup-tool/pull/11/commits/821613f848c7af8c3a4658b0516811f992c6a7a9)

Some more information though: The current approach for sleep hooks is not an official or recommended method for running scripts. As seen on man pages: https://www.freedesktop.org/software/systemd/man/systemd-suspend.service.html it is a hack. In addition, the scripts approach requires logging to syslog, which is harder to parse from extension than using systemd services which log to journal.  

Link hook function is useful for linking service files and daemon-reload systemd. Which is why I included it for now, it is used in the PR this is split from: https://github.com/microsoft/hibernation-setup-tool/pull/7